clients/src/main/java/org/apache/kafka/common/protocol/types/ArrayOf.java

```diff
@@ -72,6 +94,9 @@ public String toString() {
     @Override
     public Object[] validate(Object item) {
         try {
+            if (isNullable() && item == null)
+                return null;
```

Comment:
Do we want to throw an exception with appropriate message if item if null but shouldn't be?